### PR TITLE
K8SPSMDB-287 document vault

### PR DESCRIPTION
**Pull request by fiowro**

### CODE REVIEW

File: docs/encryption.md [hunk 1]
```diff
@@ -19,17 +21,151 @@ edited to turn this feature on:
...
+    ```bash
+    $ kubectl exec -it pod/vault-service-0 -- vault operator init -key-shares=1 -key-threshold=1 -format=json > /tmp/vault-init
+    $ unsealKey=$(jq -r ".unseal_keys_b64[]" < /tmp/vault-init)
+    ```
+
+    To unseal Vault, execute the following command **for each Pod** of Vault
+    running:
+
+    ```bash
+    $ kubectl exec -it pod/vault-service-0 -- vault operator unseal "$unsealKey"
```
Comment (spron-in) [inline docs/encryption.md:71]: In the latest version of a helm chart it is `vault-0`, not `vault-service-0`.

File: docs/encryption.md [hunk 2]
```diff
@@ -19,17 +21,151 @@ edited to turn this feature on:
...
+        kind: Secret
+        metadata:
+          name: my-cluster-name-vault
+        type: Opaque
+        stringData:
+          keyring_vault.conf: |-
+            token = = s.VgQvaXl8xGFO1RUxAPbPbsfN
+            vault_url = https://vault-service.vault-service.svc.cluster.local
+            secret_mount_point = psmdb-secret
+            vault_ca = /etc/mysql/vault-keyring-secret/ca.cert
```
Comment (spron-in) [inline docs/encryption.md:150]: Copy paste from MySQL. Wrong path here.

File: docs/encryption.md [hunk 3]
```diff
@@ -19,17 +21,151 @@ edited to turn this feature on:
...
+        !!! note
+
+            the `name` key in the above file should be equal to the
+            `secrets.vault` key from the `deploy/cr.yaml` configuration
+            file.
+
+    2. To turn on TLS and access the Vault server via HTTPS, you should do two more things:
+
+        * add one more item to the secret: the contents of the `ca.cert` file
+            with your certificate,
```
Comment (spron-in) [inline docs/encryption.md:136]: where do I get this ca.cert?

**Review by spron-in — CHANGES_REQUESTED** [review]

File: docs/encryption.md [hunk 4]
```diff
@@ -19,17 +21,151 @@ edited to turn this feature on:
...
+        ```yaml
+        apiVersion: v1
+        kind: Secret
+        metadata:
+          name: my-cluster-name-vault
+        type: Opaque
+        stringData:
+          keyring_vault.conf: |-
+             token = s.VgQvaXl8xGFO1RUxAPbPbsfN
+             vault_url = vault-service.vault-service.svc.cluster.local
```
Comment (spron-in) [inline docs/encryption.md:123]: this is also wrong.
1. `vault-service` will be just `vault`
2. `vault-service.vault-service.svc.cluster.local` implies that you are installing vault in `vault-service` namespace which is not done in the examples above.

So it should be smth like `vault.default.svc.cluster.local` if the user follows the instruction.

**Review by spron-in — CHANGES_REQUESTED** [review]

File: docs/encryption.md [hunk 5]
```diff
@@ -5,31 +5,183 @@ is supported by the Operator since version 1.1.0.
...
+
+    ```text
+    s.VgQvaXl8xGFO1RUxAPbPbsfN
+    ```
+
+    Now login to Vault with this token and enable the key-value secret engine:
+
+    ```bash
+    $ kubectl exec -it vault-0 -- /bin/sh
+    $ vault login s.VgQvaXl8xGFO1RUxAPbPbsfN
```
Comment (nmarukovich) [inline docs/encryption.md:124]: We may add the command output @spron-in what do you think? 
`marukovich@Natalias-MacBook-Pro percona-version-service [psmdb-1.13.0_update_operator_image] $ kubectl exec -it vault-0 -- /bin/sh
/ $ vault login hvs.7hTVYdJjROkpDJAxSLl3HKH9
Success! You are now authenticated. The token information displayed below
is already stored in the token helper. You do NOT need to run "vault login"
again. Future Vault requests will automatically use this token.

Key                  Value
---                  -----
token                hvs.7hTVYdJjROkpDJAxSLl3HKH9
token_accessor       iMGp477aReYkPBWrR42Z3L6R
token_duration       ∞
token_renewable      false
token_policies       ["root"]
identity_policies    []
policies             ["root"]`

File: docs/encryption.md [hunk 6]
```diff
@@ -5,31 +5,183 @@ is supported by the Operator since version 1.1.0.
...
+    $ vault secrets enable -path secret kv-v2
+    ```
```
Comment (nmarukovich) [inline docs/encryption.md:126]: `Success! Enabled the kv-v2 secrets engine at: secret/`

File: docs/encryption.md [hunk 7]
```diff
@@ -5,31 +5,183 @@ is supported by the Operator since version 1.1.0.
...
+
+    !!! note
+
+        You can also enable audit, which is not mandatory, but useful:
+
+        ```bash
+        $ vault audit enable file file_path=/vault/vault-audit.log
+        ```
```
Comment (nmarukovich) [inline docs/encryption.md:134]: `Success! Enabled the file audit device at: file/`

**Comment by nmarukovich** [comment]
If we want to use tls we need to 
1. generate secret 
kubectl create secret generic vault-secret --from-literal=token="s.VgQvaXl8xGFO1RUxAPbPbsfN" --from-file=ca.crt=<path>/to/ca.crt
2. Add value to config 
tokenFile: /etc/mongodb-vault/token
serverCAFile: /etc/mongodb-vault/ca.crt

and it should work

**Review by nmarukovich — APPROVED** [review]

File: docs/encryption.md [hunk 8]
```diff
@@ -168,34 +168,67 @@ The following steps will deploy Vault on Kubernetes with the [Helm 3 package man
...
         ...
-    ```
+        configuration: |
+          ...
+          security:
+            enableEncryption: true
+            vault:
+              serverName: vault
+              port: 8200
+              tokenFile: /etc/mongodb-vault/token
```
Comment (hors) [inline docs/encryption.md:226]: two times `tokenFile`

**Review by hors — CHANGES_REQUESTED** [review]

**Review by hors — APPROVED** [review]

**Pull request merged**